pkg/replication/replicator.go

```diff
@@ -164,9 +163,11 @@ func (r replicator) initializeReplicationWithCheck(
 	// To make a closed channel
 	close(replication.enactedCh)
 
-	err = replication.lockHosts(overrideLock, r.lockMessage)
-	if err != nil {
-		return nil, errCh, err
+	if !skipLocking {
+		err = replication.lockHosts(overrideLock, r.lockMessage)
```

Comment:
saved you a click: this function really creates a lock based on podID. It has nothing to do with hosts anymore